lib/http_clients/scr_authorizer.ex

```diff
@@ -0,0 +1,52 @@
+defmodule HttpClients.ScrAuthorizer do
+  @moduledoc """
+  Client for SCR Authorizer calls
+  """
+  alias HttpClients.ScrAuthorizer.ProponentAuthorization
+
+  @spec create_proponent_authorization(Tesla.Client.t(), ProponentAuthorization.t()) ::
+          {:error, any} | {:ok, Proponent.t()}
+  def create_proponent_authorization(
+        %Tesla.Client{} = client,
+        %ProponentAuthorization{} = authorization
+      ) do
+    case Tesla.post(client, "/v1/proponent-authorizations", authorization) do
+      {:ok, %Tesla.Env{status: 201} = response} -> {:ok, build_struct(response.body["data"])}
+      {:ok, %Tesla.Env{} = response} -> {:error, response}
+      {:error, reason} -> {:error, reason}
+    end
+  end
+
+  defp build_struct(proponent_authorization) do
```

Comment:
```suggestion
  defp build_proponent_authorization(authorization) do
```
:nail_care: que tal? achei `build_struct` mto genérico

Reply:
boa, prefiro tbm!